_posts/2020-09-11-micro-1-0-0-released/index.md

```diff
@@ -0,0 +1,86 @@
+* How many of them were successfully processed vs ended up as "bad" (e.g. because the events failed validation against the corresponding schemas in the Iglu Schema Registry)
+* For any events that have successfully processed, what type of events they are, what fields have been recorded etc.
+* For any events that have not been successfully processed, what errors were generated on processing the events. (So these can be surfaced back via the test suite.)
+* This means companies can build automated test suites to ensure that specific events in an application generate specific events that are successfully processed by Snowplow.
+
+For examples of how to use Snowplow Micro, take a look at the 'How to' guides for [Cypress](https://snowplowanalytics.com/blog/2020/08/07/testing-tracking-with-micro-and-cypress/) and [Nightwatch](https://snowplowanalytics.com/blog/2020/08/07/testing-tracking-with-micro-and-nightwatch/) as well as the [examples repository](https://github.com/snowplow-incubator/snowplow-micro-examples#snowplow-micro-examples). 
+
+
+Read on below for:
+
+1. [Breaking changes since version 0.1.0](#breaking-changes)
+2. [Updates to the validation logic](#validation)
+3. [Other updates and bug fixes](#updates)
+4. [Documentation and help](#documentation-and-help)
+
+<!--more-->
+
+<h2 id="breaking-changes">1. Breaking changes since version 0.1.0</h2>
+
+__The good event format has changed__
+
+The `/micro/good` endpoint returns a JSON array of good events. The structure of the JSON objects in this array has changed ([GitHub issue #28](https://github.com/snowplow-incubator/snowplow-micro/issues/28)). 
+
+In version 0.1.0, each `GoodEvent` used to contain the following four fields:
+
+* `event`: the `RawEvent`, in the format of a validated event before enrichment
+* `eventType`: the type of the event
+* `schema`: the schema of the event in the case of an unstructured event
+* `contexts`: the contexts attached to the event
+
+In version 1.0.0, each `GoodEvent` now contains the following four fields:
+
+* `rawEvent`: the `RawEvent`, in the format of a validated event before enrichment (previously `event`)
+* `event`: the Canonical Snowplow Event, in the format of a validated event after enrichment (with all the enrichments deactivated)
+* `eventType`: the type of the event
+* `schema`: the schema of the event in the case of an unstructured event
+* `contexts`: the contexts attached to the event
+
+__The format of the configuration file has changed__
+
+In the configuration file, variables that were previously camel case now use hyphens. For example, `crossDomain` [here](https://github.com/snowplow-incubator/snowplow-micro/blob/micro-0.1.0/example/micro.conf#L34) has become `cross-domain` [here](https://github.com/snowplow-incubator/snowplow-micro/blob/master/example/micro.conf#L41). Therefore, when switching from Snowplow Micro version 0.1.0 to 1.0.0 we recommend you copy the latest example config from the GitHub repository.
```

Comment:
Hey @benjben can you comment here about why the change from camel case to hyphens?  Does it make the config style compatible with the stream collector now?

This is going to be an annoying change for existing users (coupled with the fact that the `latest` tag has changed) so it would be good to explain why this has happened.

Reply:
Hey @istreeter, I've realized that the change in configuration you mentioned is not the same as I thought. I was thinking that it was the same as [here](https://github.com/snowplow-incubator/snowplow-micro/commit/9001a79cdc502afe71bf962ea4f573c4a0a15996#diff-d0091eb64a019c3fd2ec1924809e8b6aL22-R19) and was normal, but you're right it actually makes Micro config and Collector config incompatible. This was introduced in the big bump of dependencies, [these lines](https://github.com/snowplow-incubator/snowplow-micro/commit/9001a79cdc502afe71bf962ea4f573c4a0a15996#diff-d0091eb64a019c3fd2ec1924809e8b6aL22-R19) and [these lines](https://github.com/snowplow-incubator/snowplow-micro/commit/9001a79cdc502afe71bf962ea4f573c4a0a15996#diff-d0091eb64a019c3fd2ec1924809e8b6aL86-L90) regarding the parsing of configuration have evolved. I think that there is a way to make it work as before. If it's not too late, we could maybe release 1.0.0 with the fix and update the blog post with 2.0.0 ?